l10n_es_aeat_mod347/models/mod347.py

```diff
@@ -49,213 +31,244 @@ def _get_default_address(self, partner):
+                        vals['amount'] = 0.0
+                        vals['cash_amount'] = amount
+                        vals['origin_fiscalyear_id'] = fy_id
+                        partner_record = partner_record_obj.create(vals)
+                        if partner_record:
+                            records.append(partner_record)
+                    else:
+                        invoice_record.write({
+                            'cash_amount': amount,
+                            'origin_fiscalyear_id': fy_id,
+                        })
+                        partner_record = invoice_record
+                    for line in cash_moves[fy_id]:
+                        cash_record_obj.create({
+                            'partner_record_id': partner_record.id,
+                            'move_line_id': line.id,
+                            'date': line.date,
+                            'amount': line.credit,
+                        })
+        return records
+
+    def _partner_records_create(self, data):
+        partner = data.get('partner')
+        address = self._get_default_address(partner)
+        partner_country_code, partner_vat = (
+            re.match(r"([A-Z]{0,2})(.*)", partner.vat or '').groups())
+        community_vat = ''
+        if not partner_country_code:
+            partner_country_code = address.country_id.code
+        partner_state_code = address.state_id.code
+        if partner_country_code != 'ES':
+            partner_vat = ''
+            community_vat = partner.vat
+            partner_state_code = 99
+        vals = {
+            'report_id': self.id,
+            'partner_id': partner.id,
+            'partner_vat': partner_vat,
+            'representative_vat': '',
+            'community_vat': community_vat,
+            'partner_state_code': partner_state_code,
+            'partner_country_code': partner_country_code,
+        }
+        # Create A record
+        self._partner_record_a_create(data, vals)
+        # Create B records
+        self._partner_record_b_create(data, vals)
+        return True
+
+    def _invoices_read(self, partners):
+        invoice_obj = self.env['account.invoice']
+        partner_obj = self.env['res.partner']
+        domain = [('state', 'in', ['open', 'paid']),
+                  ('period_id', 'in', self.periods.ids),
+                  ('not_in_mod347', '=', False)]
+        groups = invoice_obj.read_group(
+            domain, ['commercial_partner_id'], ['commercial_partner_id'])
+        key_field = 'id'
+        invoice_types = ['in_invoice']
+        refund_types = ['in_refund']
+        if not self.only_supplier:
+            invoice_types.append('out_invoice')
+            refund_types.append('out_refund')
+        if self.group_by_vat:
+            key_field = 'vat'
+        for group in groups:
+            partner = partner_obj.browse(group['commercial_partner_id'][0])
+            if not partner or partner.not_in_mod347:
```

Comment:
Si te devuelve ese dato Odoo, es que existe, por lo que no hay necesidad del `if not partner`. De todas formas, para comprobar la existencia de un registro, no se hace así, si no que hay que hacer `if not partner.exists()`, porque lo otro te puede devolver True pero no existir (tú puedes hacer browse de cualquier ID, y te devuelve un recordset no vacío), pero al acceder a un dato tendrás el error de "Missing...")